### ISSUE TITLE
Consolidate differences in the JIT for crossgen and runtime scenarios

### ISSUE THREAD

**Issue by gkhanna79** [dotnet/coreclr#4832]
src\jit\crossgen produces a version of clrjit_crossgen that is built with different defines (it includes crossgen.cmake) than the standalone jit we intend to use for RTM from src\jit\standalone (which only defines FEATURE_READYTORUN_COMPILER). 

This needs to be consolidated to ensure that there is always one jit for all scenarios (crossgen vs runtime) - unless we want to have two different JIT binaries.

@russellhadley @pgavlin PTAL

**Comment by gkhanna79** [comment]
CC @fadimounir @jkotas

**Comment by pgavlin** [comment]
I looked through the JIT for uses of the symbols that are not defined by the standalone build but are defined by crossgen.cmake (the standalone build defines `FEATURE_READYTORUN_COMPILER` and `SELF_NO_HOST`). The only missing symbol that is used by the JIT is `CROSSGEN_COMPILE`, and that is only used to control logging and noway asserts:
```
compiler.cpp:#ifndef CROSSGEN_COMPILE
error.cpp:#if defined(CROSSGEN_COMPILE) && !defined(PLATFORM_UNIX)
error.cpp:#else // CROSSGEN_COMPILE
error.cpp:#endif // CROSSGEN_COMPILE
host.h:#if defined(CROSSGEN_COMPILE) && !defined(PLATFORM_UNIX) && !defined(fprintf)
jitconfigvalues.h:#if (!defined(DEBUG) && !defined(_DEBUG)) || (defined(CROSSGEN_COMPILE) && !defined(FEATURE_CORECLR))
jitconfigvalues.h:#else // (defined(DEBUG) || defined(_DEBUG)) && (!defined(CROSSGEN_COMPILE) || defined(FEATURE_CORECLR))
jitconfigvalues.h:#endif // (!defined(DEBUG) && !defined(_DEBUG)) || (defined(CROSSGEN_COMPILE) && !defined(FEATURE_CORECLR))
```

The changes to logging are there to accommodate the fact that crossgen sets the console to support only Unicode output, and we'll need to figure out an alternative strategy for detecting that situation. The config value disables noway asserts for Desktop crossgen; we'll need to think about what to do here.

cc @BruceForstall

**Comment by pgavlin** [comment]
Fixed by #4942.
